pkgs/development/python-modules/outlines-core/default.nix

```
@@ -0,0 +1,78 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  cargo,
+  rustPlatform,
+  rustc,
+  setuptools-rust,
+  setuptools-scm,
+  interegular,
+  jsonschema,
+  datasets,
+  numpy,
+  pytestCheckHook,
+  pydantic,
+  scipy,
+  torch,
+  transformers,
+}:
+
+buildPythonPackage rec {
+  pname = "outlines-core";
+  version = "0.1.17";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "outlines_core";
+    hash = "sha256-CSWhKrHogYxvGJb+79tNTYySLmRhUVBi0b0+YEWbAoE=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  postPatch = ''
+    ls -l
+    cat ${./Cargo.lock} > Cargo.lock
+    ls -l
```

```suggestion
    ln -s ${./Cargo.lock} Cargo.lock
```
https://nixos.org/manual/nixpkgs/stable/#importing-a-cargo.lock-file

---

Does it work with `-sf`?
If not, I would recommend just using `cp` here instead. `cat`ing a file like this is very odd

---

> Does it work with `-sf`?

Doesn't work, the issue is that the linked-to file is read-only. Any kind of symlink will be read-only as a result.

> If not, I would recommend just using `cp` here instead. `cat`ing a file like this is very odd

Doesn't work, because the permissions are copied and the file will be read-only. What would probably work is something like `cp --no-preserve=mode`, but at that point it's longer than using `cat`. The cat pattern is not that odd, it's used more in nixpkgs:

```
git grep "cat \$.*>"
```

It's just an easy way to recreate a file with the standard umask/uid/gid. At any rate, change it to `cp --no-preserve=mode`.